### ISSUE TITLE
How can I send an email when the SMTP domain it was sent from doesn't match the domain in the "From:"?

### ISSUE THREAD

**Issue by kris18shadow** [bbottema/simple-java-mail#459]
I am getting the following exceptioin:  
```
 javax.mail.SendFailedException: Invalid Addresses; 
nested exception is: com.sun.mail.smtp.SMTPAddressFailedException: 553 5.7.1 <test@example.com>: 
Sender address rejected: not owned by user admin@domain.com 
```

when I am trying to send an email which "from" differs from the username of the SMTP I am using. It's obvious why it fails in this case I have done it before with the quarkus mailer, but now I have switched to simple-java-mail (version 6.7.6). Is there a way that I can achieve this?

For example in gmail, the email should have a "via" in the from box like this: 
![image](https://user-images.githubusercontent.com/7622327/233780994-9a6b0053-866c-4a26-8bb4-e83ce6b47b21.png)

In my particular case from above it should look like this: **test@example.com via domain.com** 
and mailed-by: **domain.com**

Thanks in advance!

**Comment by bbottema** [comment]
Seems like a server policy to disallow from addresses that don't match the username. I'm not sure what I can do about this.

**Comment by bbottema** [comment]
I did some research and I found we can't force the server to ignore its policy, but we can try to appease it by setting the "Sender" header on the envelope.

Before I add this as a nice API, could you please try using the following example:

```java
currentMailerBuilder
    .withHeader("Sender", "admin@domain.com")
```

**Comment by kris18shadow** [comment]
Hi, thanks for replying! 

I tried your suggestion and sadly the result is the same. I can send the mail succesfully only if I set the `sender` header to be  the test@example.com and leave `from` as admin@domain.com , but of course this does not result in the desired behavior from my example screenshot above.

In the quarkus mailer case I achieve it like this:
```
Mail.withHtml(to, subject, html)
                .addHeader("From", fromHeader)
                .addHeader("Reply-To", fromHeader);
```
Given that it is the exact same case (same domain, SMTP server, user, "from" address and recipient, etc)
So basically I add a "from" header manually which differs from the domain of my SMTP. It seems like somehow the quarkus mailer gets around this server policy (it could be a security issue on the SMTP server side as far as I know).

I tried adding a "from" header manually in simple-java-mail too, but I cannot build a valid email without specifiying a value for `from()` in the `EmailBuilder`, and of course if I add both `from()` and then add manually a "from" header - gmail refuses to deliver the message with this:
```
 host gmail-smtp-in.l.google.com
    said: 550-5.7.1  Our system has detected that this message
    is not RFC 550-5.7.1 5322 compliant: 550-5.7.1 Multiple 'From' headers
    found. 550-5.7.1 To reduce the amount of spam sent to Gmail, this message
    has been 550-5.7.1 blocked.
```

If you have any ideas - they will would be highly aprreciated, meanwhile if I find out what's the difference between the outputs of the quarkus mailer and simple-java-mail - I will let you know.
Thank you!
